### PR TITLE
ci: Start typechecking `n8n-workflow` and `n8n-core` (no-changelog)

### PR DESCRIPTION
**Pull request by netroy**
## Summary

Now only the `cli` package remains excluded from type-checking on the CI.

## Review / Merge checklist

- [x] PR title and summary are descriptive

### CODE REVIEW

File: packages/workflow/test/FilterParameter.test.ts [hunk 1]
```diff
@@ -1035,163 +1044,187 @@ describe('FilterParameter', () => {
...
-									leftValue: left,
-									rightValue: right,
-									operator: { operation: 'contains', type: 'array', rightType: 'any' },
-								},
-							],
-						}),
-					);
-					expect(result).toBe(expected);
-				});
+				] as unknown as Tests)(
```
Comment (despairblue) [inline packages/workflow/test/FilterParameter.test.ts:1047]: Would changing the definition of `FilterConditionValue` be wrong?

```ts
export type FilterConditionValue = {
	id: string;
	leftValue: NodeParameterValue | NodeParameterValue[];
	operator: FilterOperatorValue;
	rightValue: NodeParameterValue | NodeParameterValue[];
};
```

Comment (netroy) [reply packages/workflow/test/FilterParameter.test.ts:1047]: i tried that and it made the editor throw errors. I want to revisit these types, but I don't think I want to do that in this PR.

Comment (netroy) [reply packages/workflow/test/FilterParameter.test.ts:1047]: I tried `NodeParameterValue | NodeParameterValue[]` and `NodeParameterValue | NodeParameterValue[] | Date | DateTime`, and both break the editor.

**Review by despairblue — DISMISSED** [review]

File: packages/workflow/test/FilterParameter.test.ts (hunk 1 above)
Comment (despairblue) [reply packages/workflow/test/FilterParameter.test.ts:1047]: Created a follow up: https://github.com/n8n-io/n8n/pull/9930

**Review by despairblue — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.50.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.50.0)

**Pull request merged**